experimental/rockbuilder/rockbuilder.py

```diff
@@ -163,7 +163,52 @@ def printout_build_arguments(args):
     print("Projects:", args.project)
 
 
-def verify_build_env(args):
+def verify_build_env__python(self):
+    # check the python used. It needs to be by default an virtual env but
+    # this can be overriden to real python version by setting up ENV variable
+    # ROCK_PYTHON_PATH
+    python_home_dir = os.path.dirname(sys.executable)
+    if "VIRTUAL_ENV" in os.environ:
+        os.environ["ROCK_PYTHON_PATH"] = python_home_dir
+    else:
+        if "ROCK_PYTHON_PATH" in os.environ:
+            if not os.path.abspath(python_home_dir) == os.path.abspath(
+                os.environ["ROCK_PYTHON_PATH"]
+            ):
+                print("Error, virtual python environment is not active and")
+                print(
+                    "PYTHON location is different than specified by the ROCK_PYTHON_PATH"
+                )
```

Comment:
What purpose does this code serve? I'm skeptical whenever a script starts making any assumptions about the Python environment. Developers can choose to run inside or outside of a virtual environment. Notably in Docker containers or ephemeral CI VMs, they will often run using the system Python. It is safer and more flexible to leave virtual environment handling up to the caller of the script.